src/qibolab/pulses.py

```diff
@@ -1304,14 +1310,16 @@ def global_phase(self):
     def copy(self):  # -> Pulse|ReadoutPulse|DrivePulse|FluxPulse:
         """Returns a new Pulse object with the same attributes."""
 
+        import copy
```

Comment:
I know this is coming from the other PR but generally top level imports (at the beginning of the file) are preferred over importing inside functions. Unless there is a special reason, eg. optional dependency, but this is not the case here since `copy` is standard library.